pkg/controller/build/image_build_request.go

```diff
@@ -311,183 +252,162 @@ func (i ImageBuildRequest) toPodmanPod() *corev1.Pod {
+			Name: "base-image-pull-creds",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: i.MachineOSConfig.Spec.BuildInputs.BaseImagePullSecret.Name,
+					Items: []corev1.KeyToPath{
+						{
+							Key:  corev1.DockerConfigJsonKey,
+							Path: "config.json",
 						},
 					},
 				},
-				{
-					// Provides a way for the "image-build" container to signal that it
-					// finished so that the "wait-for-done" container can retrieve the
-					// iamge SHA.
-					Name: "done",
-					VolumeSource: corev1.VolumeSource{
-						EmptyDir: &corev1.EmptyDirVolumeSource{
-							Medium: corev1.StorageMediumMemory,
+			},
+		},
+		{
+			// Provides the credentials needed to push the final OS image.
+			Name: "final-image-push-creds",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: i.MachineOSConfig.Spec.BuildInputs.RenderedImagePushSecret.Name,
+					Items: []corev1.KeyToPath{
+						{
+							Key:  corev1.DockerConfigJsonKey,
+							Path: "config.json",
 						},
 					},
 				},
 			},
 		},
-	}
-}
-
-// We're able to run the Buildah image in an unprivileged pod provided that the
-// machine-os-builder service account has the anyuid security constraint
-// context enabled to allow us to use UID 1000, which maps to the UID within
-// the official Buildah image.
-// nolint:dupl // I don't want to deduplicate this yet since there are still some unknowns.
-func (i ImageBuildRequest) toBuildahPod() *corev1.Pod {
-	env := []corev1.EnvVar{
-		{
-			Name:  "DIGEST_CONFIGMAP_NAME",
-			Value: i.getDigestConfigMapName(),
-		},
-		{
-			Name:  "HOME",
-			Value: "/home/build",
-		},
-		{
-			Name:  "TAG",
-			Value: i.FinalImage.Pullspec,
-		},
 		{
-			Name:  "BASE_IMAGE_PULL_CREDS",
-			Value: "/tmp/base-image-pull-creds/config.json",
+			// Provides a way for the "image-build" container to signal that it
+			// finished so that the "wait-for-done" container can retrieve the
+			// iamge SHA.
+			Name: "done",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{
+					Medium: corev1.StorageMediumMemory,
+				},
+			},
 		},
 		{
-			Name:  "FINAL_IMAGE_PUSH_CREDS",
-			Value: "/tmp/final-image-push-creds/config.json",
+			// This provides a dedicated place for Buildah to store / cache its
+			// images during the build. This seems to be required for the build-time
+			// volume mounts to work correctly, most likely due to an issue with
+			// SELinux that I have yet to figure out. Despite being called a cache
+			// directory, it gets removed whenever the build pod exits
+			Name: "buildah-cache",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
 		},
 	}
 
-	var uid int64 = 1000
-	var gid int64 = 1000
-
-	securityContext := &corev1.SecurityContext{
-		RunAsUser:  &uid,
-		RunAsGroup: &gid,
+	// Octal: 0755.
+	var mountMode int32 = 493
```

Comment:
note (non-blocking): One can use an octal literal by doing something like `var mountMode int32 = 0o755`. Not sure if that would work here, but it's just a thought.

(This does not need to be addressed right now though.)